homeassistant/components/person/__init__.py

```diff
@@ -0,0 +1,145 @@
+"""
+Support for tracking people.
+
+For more details about this component, please refer to the documentation.
+https://home-assistant.io/components/person/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.device_tracker import (
+    DOMAIN as DEVICE_TRACKER_DOMAIN)
+from homeassistant.const import (
+    ATTR_ID, ATTR_LATITUDE, ATTR_LONGITUDE, CONF_ID, CONF_NAME)
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.helpers.event import async_track_state_change
+from homeassistant.helpers.restore_state import RestoreEntity
+
+_LOGGER = logging.getLogger(__name__)
+ATTR_SOURCE = 'source'
+ATTR_USER_ID = 'user_id'
+CONF_DEVICE_TRACKERS = 'device_trackers'
+CONF_USER_ID = 'user_id'
+DOMAIN = 'person'
+
+PERSON_SCHEMA = vol.Schema({
+    vol.Required(CONF_ID): cv.string,
+    vol.Required(CONF_NAME): cv.string,
+    vol.Optional(CONF_USER_ID): cv.string,
+    vol.Optional(CONF_DEVICE_TRACKERS, default=[]): vol.All(
+        cv.ensure_list, cv.entities_domain(DEVICE_TRACKER_DOMAIN)),
+})
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.All(cv.ensure_list, [PERSON_SCHEMA])
+}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup(hass, config):
+    """Set up the person component."""
+    component = EntityComponent(_LOGGER, DOMAIN, hass)
+    conf = config[DOMAIN]
+    entities = []
+    for person_conf in conf:
+        user_id = person_conf.get(CONF_USER_ID)
+        if (user_id is not None
+                and await hass.auth.async_get_user(user_id) is None):
```

Comment:
Should we skip it completely or just let it be added without a user_id ?

Reply:
Not sure. I can see arguments for both ways.